### PR TITLE
server: fix column length when convert column info for tinyint

### PR DESCRIPTION
**Pull request by zhexuany**
When column type is `tinyint`, [ao](https://sourcegraph.com/github.com/Kast0rTr0y/ao@189a44ab341e5d613ae9db0e98365688b80162ed/-/blob/activeobjects-core/src/net/java/ao/types/TypeManager.java#L258) checks both jdbc type and its precision in qualifiers. It needs precision in qualifiers to be 1 but tidb returned column's precision is 4(here precision is just column length). This PR fixes this. 

@zz-jason @XuHuaiyu PTAL and please check whether navicat has same behavior.

### CODE REVIEW

**Comment by zz-jason** [comment]
LGTM

File: server/driver_tidb_test.go [hunk 1]
```diff
@@ -60,4 +60,38 @@ func (ts tidbResultSetTestSuite) TestConvertColumnInfo(c *C) {
...
+				Collate: charset.CollationUTF8,
+			},
+			Comment: "column a is the first column in table dual",
+		},
+		ColumnAsName: model.NewCIStr("a"),
+		TableAsName:  model.NewCIStr("dual"),
+		DBName:       model.NewCIStr("test"),
+	}
+	colInfo = convertColumnInfo(&resultField)
+	c.Assert(colInfo.Schema, Equals, "test")
```
Comment (zz-jason) [inline server/driver_tidb_test.go]: how about using `DeepEquals` ?

**Comment by shenli** [comment]
Good job!
LGTM

**Comment by zhexuany** [comment]
@zz-jason @shenli PTAL again.

**Pull request merged**